include/seqan3/alignment/pairwise/detail/type_traits.hpp

```diff
@@ -83,6 +85,12 @@ template <typename configuration_t>
 struct alignment_configuration_traits
 {
 private:
+    //!\brief Transforms integral types to unsigned types and chooses the minimal viable uint for floating point types.
+    template <arithmetic score_t>
+    using select_scalar_index_t = lazy_conditional_t<std::integral<score_t>,
+                                                     lazy<std::make_unsigned_t, score_t>,
+                                                     min_viable_uint_t<1ull << (sizeof_bits<score_t> - 1)>>;
```

Comment:
Why? This is really not making it better inside of the class. Also it would be:

```
    template <arithmetic score_t>
    using select_scalar_index_t = lazy_conditional_t<std::integral<score_t>,
                                                     lazy<unsigned_scalar_index_of_integral, score_t>,
                                                     lazy<unsigned_scalar_index_of_floating_point, score_t>;
```
If you like I can add a more descriptive comment or detail body to the type.